.github/workflows/build.yml

```
@@ -64,75 +64,148 @@ jobs:
       - name: Create Amplify branch
-        if: github.event_name == 'pull_request' && inputs.deploy == true
+        if: inputs.deploy == true
         shell: bash
         # We return true to prevent the step from failing if the branch already exists
         run: |
-            aws amplify create-branch \
-            --app-id ${{ inputs.amplify-app-id }} \
+          aws amplify create-branch \
+            --app-id ${{ env.AMPLIFY_ID }} \
             --branch-name ${{ env.BRANCH_NAME }} \
             || true
 
       # Deploy Amplify from S3
       - name: Deploy Amplify from S3
-        if: github.event_name == 'pull_request' && inputs.deploy == true
+        if: inputs.deploy
         shell: bash
         run: |
-            aws amplify start-deployment \
-            --app-id ${{ inputs.amplify-app-id }} \
+          aws amplify start-deployment \
+            --app-id ${{ env.AMPLIFY_ID }} \
             --branch-name ${{ env.BRANCH_NAME }} \
             --source-url https://${{ env.BUCKET_NAME }}.s3.${{ env.AWS_REGION }}.amazonaws.com/${{ env.ZIP_NAME }}
+      # Set env vars for domain name association
+      - name: Set environment variables for domain name association
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          PACKAGE_NAME_SUFFIX=$(echo "${{ inputs.package-name }}" | sed -e 's/^-app//')
+          echo "SUB_DOMAIN=pr${{ github.event.number }}-aperture-$PACKAGE_NAME_SUFFIX" >> $GITHUB_ENV
+
+      - name: Get existing domain associations
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          DOMAIN_ASSOCIATIONS=$(aws amplify get-domain-association \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --domain-name pie.design \
+            --region ${{ env.AWS_REGION }} \
+            | jq '
+              [.domainAssociation.subDomains[].subDomainSetting | { prefix: (.prefix // ""), branchName }] |
+              . |= (map(.branchName) | index("pr${{ github.event.number }}")) as $ix |
+              if $ix == null then
+                . + [{branchName: "pr${{ github.event.number }}", prefix: "${{ env.SUB_DOMAIN }}"}]
+              else . end
+            ' -c)
+          echo "DOMAIN_ASSOCIATIONS=$DOMAIN_ASSOCIATIONS" >> $GITHUB_ENV
+      - name: Create Subdomain Association
+        if: github.event_name == 'pull_request' && inputs.deploy
+        shell: bash
+        run: |
+          aws amplify update-domain-association \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --domain-name pie.design \
+            --sub-domain-settings '${{ env.DOMAIN_ASSOCIATIONS }}'
+
+
+      - name: Wait for Amplify deployment
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          LATEST_JOB_ID=$(aws amplify list-jobs \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --branch-name ${{ env.BRANCH_NAME }} \
+            --region ${{ env.AWS_REGION }} \
+            --output text)
```

I think you're missing the query here to extract the id
```suggestion
          LATEST_JOB_ID=$(aws amplify list-jobs \
            --app-id ${{ env.AMPLIFY_ID }} \
            --branch-name ${{ env.BRANCH_NAME }} \
            --region ${{ env.AWS_REGION }} \
            --query jobSummaries[0].jobId \
            --output text)
```